src/components/Button/Button.test.tsx

```diff
@@ -1,11 +1,11 @@
-import React from 'react';
+import * as React from 'react';
 import { render, screen } from '@testing-library/react';
-
+import '@testing-library/jest-dom';
 import Button from './Button';
 
 describe('Button', () => {
-  test('renders the Button component', () => {
-    render(<Button label="Hello world!" />);
-    expect(screen.getByRole('button').textContent).toContain('Hello world!');
+  it('should render a button with the class of primary', () => {
+    render(<Button>Test Button</Button>);
+    expect(screen.getByText(/Test Button/)).toBeInTheDocument();
   });
 });
```

Comment:
You could add a test to check if a button click triggers onClick correctly:
```
  const mockOnClick = jest.fn();
  it('should trigger onClick when clicked.', () => {
    render(<Button onClick={mockOnClick}>Test Button</Button>);
    userEvent.click(screen.getByText(/Test Button/));
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
```

Reply:
Thank you Tobi, appreciate the helpful comment. Done